File: 2490-circular-sentence/2490-circular-sentence.java
class Solution {
    public boolean isCircularSentence(String s) {
        int n = s.length();
        if(s.charAt(n-1) != s.charAt(0)) return false;
        
        int i=0;
        
        while(i<n) {
            while(i<n && s.charAt(i) != ' ' ) {
                i++;
            }
            if(i<n && s.charAt(i-1) != s.charAt(i+1)) return false;
            i++;
        }
        return true;
    }
}

File: 2490-circular-sentence/2490-circular-sentence.cpp
class Solution {
public:
    bool isCircularSentence(string s) {
        if(s.back() != s[0]) return false;
        
        int n = s.size();
        int i = 0;
        
        while(i<n) {
            while(i<n && s[i] != ' ') {
                i++;
            }
            
            if(i<n && s[i-1] != s[i+1]) return false;
            i++;
        }
        return true;
    }
};